core/src/solution_submission.rs

```diff
@@ -99,16 +98,32 @@ impl From<Error> for SolutionSubmissionError {
 pub struct StableXSolutionSubmitter<'a> {
     contract: &'a (dyn StableXContract + Sync),
     gas_price_estimating: &'a (dyn GasPriceEstimating + Sync),
+    retry_with_gas_price_increase: Box<dyn RetryingWithGasPriceIncrease + Send + Sync>,
 }
 
 impl<'a> StableXSolutionSubmitter<'a> {
+    #[cfg(not(test))]
+    pub fn new(
+        contract: &'a (dyn StableXContract + Sync),
+        gas_price_estimating: &'a (dyn GasPriceEstimating + Sync),
+    ) -> Self {
+        Self {
+            contract,
+            gas_price_estimating,
+            retry_with_gas_price_increase: Box::new(retry::RetryWithGasPriceIncrease {}),
+        }
+    }
+
+    #[cfg(test)]
```

Comment:
nit: Instead of conditional compilation we can have a `new` method and a `with_retrying` method. I find it a bit confusing that the same method `new` has multiple signatures. Additionally, for people unfamiliar with the codebase that may use the unit test code as a reference, it would also be unnecessarily counterintuitive.